cv_next/app/auth/callback/route.ts

```diff
@@ -2,19 +2,27 @@
 import { NextResponse } from "next/server";
 import Definitions from "@/lib/definitions";
 import SupabaseHelper from "@/server/api/supabaseHelper";
+import { checkRedirect } from "@/lib/utils";
 
 export async function GET(request: Request) {
   const { searchParams, origin } = new URL(request.url);
   const code = searchParams.get("code");
-  const next = searchParams.get("next");
+  const next = searchParams.get("next") || "";
 
   if (code) {
     const supabase = SupabaseHelper.getSupabaseInstance();
     const { error } = await supabase.auth.exchangeCodeForSession(code);
     if (!error) {
-      return NextResponse.redirect(
-        `${origin}` + Definitions.AUTH_DEFAULT_REDIRECT + next
-      );
+      if (checkRedirect(next)) {
+        return NextResponse.redirect(
+          `${origin}${Definitions.AUTH_DEFAULT_REDIRECT}${next}`
+        );
+      } else {
+        return NextResponse.json(
```

Comment:
Do we handle this somehow? Does this cause a redirect to some error page we have?

Reply:
Redirects to a black page with the text
In retrospect a bad idea - this is called only when the user deliberately tries accessing a non existing page after login so maybe we redirect to feed or 404 or something?

Reply:
404 is good